### PR TITLE
[WS1] Add CUDA and Triton SiLU/SwiGLU activation kernels

### PR DESCRIPTION
**Pull request by maxiaosong1124**
## Summary

  Adds CUDA and Triton implementations of SiLU and SwiGLU for the WS1 batch-invariant forward chain.

  The new kernels provide deterministic, element-wise forward and backward paths and match the existing PyTorch FP32 ground-truth implementation under the shared #108 numerical contract.

  This PR also wires both operators into the kernel registry and the shared operator validation harness.

  Part of #271.
  Related to #83 and #108.

  ## Implementation

  - Add native CUDA forward and backward kernels for:
    - `silu(x) = x * sigmoid(x)`
    - `swiglu(gate, up) = silu(gate) * up`
  - Add equivalent Triton forward and backward kernels.
  - Perform kernel math in FP32 and cast results back to the input dtype for the regular `forward` path.
  - Provide `forward_fp32` as the FP32 ground-truth path.
  - Support FP16, BF16, and FP32 inputs.
  - Support arbitrary tensor shapes, including non-contiguous and empty tensors.
  - Add CUDA extension bindings and type declarations.
  - Register CUDA, Triton, and PyTorch candidates in `OP_SPECS`.
  - Update backend dispatch:
    - CUDA: CUDA -> Triton -> PyTorch
    - ROCm: Triton -> PyTorch
    - CPU: PyTorch
  - Add operator documentation and usage examples.

  ## Numerical contract and invariance

  SiLU and SwiGLU are element-wise and row-independent, so no cross-row reductions or atomics are required.

  The tests verify:

  - forward and backward parity against the independent FP32 PyTorch reference;
  - bitwise batch-slice invariance;
  - bitwise padding-layout invariance for outputs and gradients;
  - deterministic repeated forward and backward execution;
  - FP16, BF16, and FP32 dtype behavior;
  - shape, dtype, and device validation for SwiGLU inputs;
  - non-contiguous and empty tensor handling;
  - CUDA and Triton integration with the shared #108 operator harness.

  For SwiGLU, `gate` and `up` must have the same shape, dtype, and device. Broadcasting is intentionally unsupported.

  ## Validation

  Validation environment:

  - NVIDIA H20
  - CUDA 12.8

  Commands and results:

  ```bash
  pre-commit run --all-files
  # all hooks passed

  python -m pytest \
    tests/test_swiglu.py \
    tests/test_op_checks.py \
    tests/test_operator_inputs.py \
    tests/test_kernel_registry.py -q
  # 142 passed, 2 skipped

  python -m pytest -q
  # 1434 passed, 120 skipped

  The CUDA extension was rebuilt successfully, and the CUDA/Triton BF16 operator-harness checks passed for both SiLU and SwiGLU.

  ## Scope

  This PR provides standalone SiLU and SwiGLU operators for WS1 correctness and consistency validation.

  Fused bias + SiLU and packed SiluAndMul variants are not included. Callers with packed gate/up tensors should split them before invoking SwiGLU.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added SiLU and SwiGLU activation operations with forward and backward support.
  - Added CUDA and Triton acceleration for supported floating-point types.
  - Added automatic backend selection with PyTorch fallback and optional FP32 execution.

- **Bug Fixes**
  - Improved validation for supported dtypes, devices, shapes, and compatible inputs.

- **Documentation**
  - Documented backend behavior, accuracy expectations, performance, and known limitations.

- **Tests**
  - Expanded coverage for gradients, dispatch, empty and noncontiguous inputs, batching, padding, and dtype validation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: tests/test_swiglu.py [hunk 1]
```diff
@@ -1,20 +1,80 @@
...
 import torch
 
+from rl_engine.kernels.gtest.op_checks import run_operator_suite
+from rl_engine.kernels.gtest.operator_specs import (
+    make_candidate,
+    make_operator_case,
+    operator_names,
+)
 from rl_engine.kernels.ops.pytorch.activation.swiglu import NativeSiLUOp, NativeSwiGLUOp
+from rl_engine.kernels.ops.triton.activation.swiglu import TritonSiLUOp, TritonSwiGLUOp
```
Comment (inaniloquentee) [inline tests/test_swiglu.py]: This top-level Triton import makes the whole file fail during collection when Triton isn’t installed. Could we guard it with a try/except and skip, like the other Triton tests?

File: tests/test_swiglu.py [hunk 2]
```diff
@@ -165,6 +236,427 @@ def test_swiglu_backward_batch_invariance_slice():
...
+
+
+def _silu_impls():
+    return [
+        "triton",
+        pytest.param("cuda", marks=requires_cuda_activation, id="cuda"),
+    ]
+
+
+@requires_nvidia_cuda
```
Comment (inaniloquentee) [inline tests/test_swiglu.py:268]: This assert feels a bit too hard for a CUDA test. If the extension isn’t built yet, this fails instead of skipping even on a valid CUDA setup. Could we make it skip unless those symbols are actually present?

**Review by inaniloquentee — CHANGES_REQUESTED** [review]

**Review by Flink-ddd — APPROVED** [review]
LGTM, Thank you.

**Review by inaniloquentee — APPROVED** [review]
